nnvm/tests/python/frontend/onnx/test_graph.py

```diff
@@ -13,13 +14,19 @@ def compare_graph(onnx_file, nnvm_sym, ishape):
     ishapes = {input_name: ishape}
     graph_attr.set_shape_inputs(g1, ishapes)
     graph_attr.set_shape_inputs(g2, ishapes)
-    g1 = g1.apply("InferShape").apply("SimplifyInference")
-    g2 = g2.apply("InferShape").apply("SimplifyInference")
+    # g1 = g1.apply("InferShape").apply("SimplifyInference")
+    # g2 = g2.apply("InferShape").apply("SimplifyInference")
```

Comment:
If I don't comment out these two lines, the tests crash at shape inference:
```
(tvm_env) hlu:tvm hlu$ python /Users/hlu/github/tvm/nnvm/tests/python/frontend/onnx/test_graph.py
Traceback (most recent call last):
  File "/Users/hlu/github/tvm/nnvm/tests/python/frontend/onnx/test_graph.py", line 31, in <module>
    test_super_resolution_example()
  File "/Users/hlu/github/tvm/nnvm/tests/python/frontend/onnx/test_graph.py", line 23, in test_super_resolution_example
    compare_graph(fname, symbol, ishape=(1, 1, 224, 224))
  File "/Users/hlu/github/tvm/nnvm/tests/python/frontend/onnx/test_graph.py", line 18, in compare_graph
    g2 = g2.apply("InferShape").apply("SimplifyInference")
  File "/Users/hlu/github/tvm/nnvm/python/nnvm/graph.py", line 234, in apply
    check_call(_LIB.NNGraphApplyPasses(self.handle, npass, cpass, ctypes.byref(ghandle)))
  File "/Users/hlu/github/tvm/nnvm/python/nnvm/_base.py", line 75, in check_call
    raise NNVMError(py_str(_LIB.NNGetLastError()))
nnvm._base.NNVMError: [14:03:31] /Users/hlu/github/tvm/nnvm/src/top/nn/convolution.cc:62: Check failed: dshape.ndim() == 4U (0 vs. 4) 62: Input data should be 4D

Stack trace returned 9 entries:
[bt] (0) 0   libnnvm_compiler.dylib              0x000000011afe9100 dmlc::StackTrace() + 288
[bt] (1) 1   libnnvm_compiler.dylib              0x000000011afe8e9f dmlc::LogMessageFatal::~LogMessageFatal() + 47
[bt] (2) 2   libnnvm_compiler.dylib              0x000000011b0d4739 nnvm::top::Conv2DInferShape(nnvm::NodeAttrs const&, std::__1::vector<nnvm::TShape, std::__1::allocator<nnvm::TShape> >*, std::__1::vector<nnvm::TShape, std::__1::allocator<nnvm::TShape> >*) + 3097
[bt] (3) 3   libnnvm_compiler.dylib              0x000000011b02cdb4 nnvm::Graph nnvm::pass::(anonymous namespace)::InferAttr<nnvm::TShape, nnvm::pass::(anonymous namespace)::$_0::operator()(nnvm::Graph) const::'lambda'(nnvm::TShape const&), std::nullptr_t>(nnvm::Graph&&, nnvm::TShape, char const*, char const*, char const*, char const*, char const*, nnvm::pass::(anonymous namespace)::$_0::operator()(nnvm::Graph) const::'lambda'(nnvm::TShape const&), std::nullptr_t)::'lambda'(unsigned int, bool)::operator()(unsigned int, bool) const + 3156
[bt] (4) 4   libnnvm_compiler.dylib              0x000000011b02adca std::__1::__function::__func<nnvm::pass::(anonymous namespace)::$_0, std::__1::allocator<nnvm::pass::(anonymous namespace)::$_0>, nnvm::Graph (nnvm::Graph)>::operator()(nnvm::Graph&&) + 3866
[bt] (5) 5   libnnvm_compiler.dylib              0x000000011affda4f nnvm::ApplyPasses(nnvm::Graph, std::__1::vector<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >, std::__1::allocator<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > > > const&) + 1407
[bt] (6) 6   libnnvm_compiler.dylib              0x000000011afe4dd6 NNGraphApplyPasses + 566
[bt] (7) 7   _ctypes.cpython-36m-darwin.so       0x000000010c0ed49f ffi_call_unix64 + 79
[bt] (8) 8   ???                                 0x00007ffee43e37a0 0x0 + 140732727703456
```
I'm very new to tvm. Do you guys know why?

Reply:
the error msg shows it is something wrong with `g2`, which is constructed by nnvm symbol and should has nothing to do with your changes?

Reply:
@yzhliu, I rebased onto master and tested again. The error went away. Must be a transient issue.